internal/cloudapi/openapi.yml

```diff
@@ -0,0 +1,240 @@
+---
+openapi: 3.0.1
+info:
+  version: '1'
+  title: OSBuild Composer cloud api
+  description: Service to build and install images.
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+
+paths:
+  /compose/{id}:
+    get:
+      summary: The status of a compose
+      parameters:
+        - in: path
+          name: id
+          schema:
+            type: string
+            format: uuid
+            example: '123e4567-e89b-12d3-a456-426655440000'
+          required: true
+          description: ID of compose status to get
+      description: Get the status of a running or completed compose. This includes whether or not it succeeded, and also meta information about the result.
+      operationId: compose_status
+      responses:
+        '200':
+          description: compose status
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeStatus'
+        '400':
+          description: Invalid compose id
+          content:
+            text/plain:
+              schema:
+                type: string
+        '404':
+          description: Unknown compose id
+          content:
+            text/plain:
+              schema:
+                type: string
+  /compose:
+    post:
+      summary: Create compose
+      description: Create a new compose, potentially consisting of several images and upload each to their destinations.
+      operationId: compose
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ComposeRequest'
+      responses:
+        '201':
+          description: Compose has started
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComposeResult'
+
+components:
+  schemas:
+    ComposeStatus:
+      required:
+        - status
+      properties:
+        status:
+          type: string
+          enum: ['success', 'failure', 'pending']
+          example: 'success'
+        image_statuses:
+          type: array
+          items:
+            $ref: '#/components/schemas/ImageStatus'
+    ImageStatus:
+      required:
+       - status
+      properties:
+        status:
+          type: string
+          enum: ['success', 'failure', 'pending', 'building', 'uploading', 'registering']
+          example: 'success'
+        upload_statuses:
+          type: array
+          items:
+            $ref: '#/components/schemas/UploadStatus'
+    UploadStatus:
+      oneOf:
+       - $ref: '#/components/schemas/AWSUploadStatus'
+    AWSUploadStatus:
```

Comment:
This is not actually used since AMI data isn't yet communicated from `awsupload.Register`.